app/src/main/java/vn/mbm/phimp/me/Camera2.java

```diff
@@ -86,6 +100,54 @@ public Camera2() {
 
 	}
 
+	private boolean inRange(int SubjectValue, int High, int Low) {
+		// In case of a 360
+		SubjectValue =  (360 + (SubjectValue % 360)) % 360;
+		if (High < Low) {
+			return High <= SubjectValue && SubjectValue <= Low;
+		} else {
+			return High <= SubjectValue || SubjectValue <= Low;
+		}
+	}
+
+	private void setFlashIcon() {
+		switch (state) {
+			case 0:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+			case 1:
+				flash.setImageResource(R.drawable.flash_off);
+				break;
+			case 2:
+				flash.setImageResource(R.drawable.flash_auto);
+				break;
+			default:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+		}
+	}
+
+	private void rotateIcons(int degrees) {
+		flash.setRotation(degrees);
+		camera_switch.setRotation(degrees);
+		buttonClick.setRotation(degrees);
+	}
+
+	private void adjustIconPositions() {
+		setFlashIcon();
+		if (inRange(camOrientation, NW, NE)) {
```

Comment:
As stated before, remove this if. Both the if and the else stars do the same thing.